src/services/auth.service.ts

```diff
@@ -0,0 +1,89 @@
+import { Injectable, HttpException, HttpStatus } from '@nestjs/common';
+import { InjectModel } from '@nestjs/mongoose';
+import jwt from 'jsonwebtoken';
+import bcrypt from 'bcrypt';
+
+import { Model } from 'mongoose';
+import { User } from '../interfaces/user.interface';
+import { SignUpDto } from 'src/dto/signup.dto';
+import { LoginDto } from 'src/dto/login.dto';
+
+@Injectable()
+export class AuthService {
+  constructor(@InjectModel('User') private readonly Users: Model<User>) {}
+
+  // To signup a user
+  async signup(signUpDto: SignUpDto): Promise<any> {
```

Comment:
Using any type here nullifies the use of TypeScript, we can define an interface for it as well.